### PR TITLE
Update hub service direct method JSON usage

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs [hunk 1]
```diff
@@ -12,13 +12,17 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public class DirectMethodServiceRequest
     {
-        private object _payload;
-
         /// <summary>
-        /// Initialize an instance of this class.
+        /// Creates an instance of this class with the specified payload to be serialized.
```
Comment (drwill-ms) [inline iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs:16]: I converted this file as close as I could to what we had before, meaning we take their object and serialize it and stuff it in a JSON type. From there we can serialize it (again!?) or offer it back to the user if they want (but why would they want it?).

This is a mostly true update, but I'd rather we change it up and just store the user payload a native object to serialize.

Thoughts?

File: iothub/service/src/DirectMethod/Models/DirectMethodClientResponse.cs [hunk 2]
```diff
@@ -28,15 +28,43 @@ protected internal DirectMethodClientResponse()
...
+            value = default;
+
+            if (JsonPayload == null)
+            {
+                return false;
+            }
+
+            try
+            {
+                value = JsonSerializer.Deserialize<T>(JsonPayload.Value.GetRawText());
```
Comment (abhipsaMisra) [inline iothub/service/src/DirectMethod/Models/DirectMethodClientResponse.cs:61]: Q - what's the difference between `JsonPayload.Value.ToString()` and `JsonPayload.Value.GetRawText()`?

File: iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs [hunk 3]
```diff
@@ -59,35 +63,19 @@ public DirectMethodServiceRequest()
...
-        public string PayloadAsJsonString { get; internal set; }
+        public string PayloadAsJsonString => JsonPayload?.GetRawText();
 
         /// <summary>
-        /// The JSON payload in JRaw type.
+        /// The deserialized payload.
         /// </summary>
         [JsonPropertyName("payload")]
-        internal JRaw JsonPayload { get; set; }
+        internal JsonElement? JsonPayload { get; set; }
```
Comment (abhipsaMisra) [inline iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs:78]: Is this the property that eventually gets serialized and sent over in the transport layer?

File: iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs:16]: > just store the user payload a native object to serialize.

This is what this PR does, right?

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**